ukis_pysat/data.py

```diff
@@ -61,23 +62,74 @@ def __init__(self, source, source_dir=None):
+        :param platform: Image platform (<enum 'Platform'>).
         :param date: Date from - to (String or Datetime tuple). Expects a tuple of (start, end), e.g.
             (yyyyMMdd, yyyy-MM-ddThh:mm:ssZ, NOW, NOW-<n>DAY(S), HOUR(S), MONTH(S), etc.)
         :param aoi: Area of interest as GeoJson file or bounding box tuple with lat lon coordinates (String, Tuple).
         :param cloud_cover: Percent cloud cover scene from - to (Integer tuple).
         :returns: Metadata of products that match query criteria (List of Metadata objects).
         """
-        if self.src == Datahub.file:
-            raise NotImplementedError("File metadata query not yet supported.")
+        if self.src == Datahub.File:
+            # query Filesystem for metadata
+            start_date = sentinelsat.format_query_date(date[0])
+            end_date = sentinelsat.format_query_date(date[1])
+            geom = self.prep_aoi(aoi)
+            if cloud_cover and platform != platform.Sentinel1:
+                min_cloud_cover = cloud_cover[0]
+                max_cloud_cover = cloud_cover[1]
+            else:
+                min_cloud_cover = 0
+                max_cloud_cover = 100
+
+            # get all json files in datadir
+            meta_files = sorted(
+                [
+                    os.path.join(dp, f)
+                    for dp, dn, filenames in os.walk(self.api)
+                    for f in filenames
+                    if f.endswith(".json")
+                ],
+                key=str.lower,
+            )
+
+            # filter json files by query parameters
+            meta_src = []
+            for meta_file in meta_files:
+                with open(meta_file) as f:
+                    try:
+                        # make sure that metadata file is valid
+                        m = json.load(f)
+                        self.construct_metadata(m)
+                        # get values from metadata
+                        m_platform = m["properties"]["platformname"]
+                        m_date = sentinelsat.format_query_date(m["properties"]["acquisitiondate"])
+                        m_geom = geometry.shape(m["geometry"])
+                        m_cloud_cover = m["properties"]["cloudcoverpercentage"]
+                        if m_cloud_cover is None:
+                            m_cloud_cover = 0
+                        if (
+                            m_platform == platform.value
+                            and m_date >= start_date
+                            and m_date < end_date
+                            and m_geom.intersects(geom)
+                            and m_cloud_cover >= min_cloud_cover
+                            and m_cloud_cover < max_cloud_cover
+                        ):
+                            meta_src.append(m)
+                    except (json.decoder.JSONDecodeError, LookupError, TypeError) as e:
+                        warnings.warn(

```

Comment:
why do we not throw an error here and remove try... except? I think such things should be handled outside of the library.

Reply:
this is a good point. then lets generally agree on moving such things outside of the library. will modify this tomorrow accordingly.